File: src/foam/lib/json/AnyKeyParser.java
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.lib.json;

import foam.lib.parse.*;

public class AnyKeyParser
  extends ProxyParser
{
  public AnyKeyParser() {
    super(new Alt(StringParser.instance(),
      new Substring(
        new Seq0(
          // TODO: Implement Java Grammar support (so separate files aren't needed)
          new IdentifierStartParser(),
          new Repeat0(new Alt(
            Range.create('0', '9'),
            new IdentifierStartParser()))))));
  }
}

File: src/foam/swift/parse/json/AnyKeyParser.js
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.CLASS({
  package: 'foam.swift.parse.json',
  name: 'AnyKeyParser',
  extends: 'foam.swift.parse.parser.ProxyParser',
  requires: [
    'foam.swift.parse.parser.Alt',
    'foam.swift.parse.parser.Literal',
    'foam.swift.parse.parser.NotChars',
    'foam.swift.parse.parser.Repeat0',
    'foam.swift.parse.parser.Seq1',
    'foam.swift.parse.parser.Substring',
  ],
  axioms: [
    foam.pattern.Singleton.create()
  ],
  properties: [
    {
      name: 'delegate',
      swiftFactory: function() {/*
return Alt_create(["parsers": [
  Seq1_create(["index": 1, "parsers": [
    Literal_create(["string": "\""]),
    Substring_create(["delegate":
      Repeat0_create(["delegate":
        NotChars_create(["chars": "\""]),
      ])
    ]),
    Literal_create(["string": "\""]),
  ]]),
  Seq1_create(["index": 0, "parsers": [
    Substring_create(["delegate":
      Repeat0_create(["delegate":
        NotChars_create(["chars": "{}, :"]),
      ])
    ])
  ]])
]])
      */},
    },
  ],
});
